toolkit/tools/imagegen/installutils/installutils.go

```diff
@@ -51,6 +52,74 @@ type PackageList struct {
 	Packages []string `json:"packages"`
 }
 
+// updatePackageRepo creates additional repo files specified by image configuration
+// and returns error if the operation fails
+func updatePackageRepo(installChroot *safechroot.Chroot, config configuration.SystemConfig) (err error) {
+
+	const (
+		repoFile     = "/etc/yum.repos.d/mariner-iso.repo"
+		squashErrors = false
+	)
+
+	if exists, ferr := file.PathExists(repoFile); ferr != nil {
+		logger.Log.Errorf("Error accessing repo file.")
+		err = ferr
+		return
+	} else if !exists {
+		err = fmt.Errorf("/etc/yum.repos.d/mariner-iso.repo does not exist")
+		return
+	}
+
+	// Loop through the PackageRepos field to determine if any customized package repos are specified.
+	// If specified, create new repo files for them and remove the default mariner-iso.repo
+	for _, packageRepo := range config.PackageRepos {
+		dstRepoPath := "/etc/yum.repos.d/mariner-" + packageRepo.Name + ".repo"
+		err = file.Copy(repoFile, dstRepoPath)
```

Comment:
What is the rational for copying the repo files rather that just making a new one? Also, the comment claims we are removing the existing repo file but it doesn't seem to be the case. (and why do we want to remove it?)

Reply:
Since the local repo already exists, I think that it would be easier to just update the fields inline compared to creating new files.

I have updated the comment. But do need to confirm with the customer to clarify what their user scenario is: pulling packages from existing repo and their repos or only from their repos during the install process.